prepare-docs.rb

```diff
@@ -5,38 +5,100 @@
 require "net/http"
 require "uri"
 require "rexml/document"
-# require "nokogiri"
+require 'json'
+require 'open-uri'
 
 $docs_repo = "https://github.com/openhab/openhab-docs"
 $docs_repo_root = $docs_repo + "/blob/main"
 $docs_repo_branch = "final"
 $addons_repo_branch = "main"
 $version = nil
+$verbose = false
 
 $ignore_addons = ['transport.modbus', 'transport.feed', 'javasound', 'webaudio', 'oh2']
 
+def verbose(message)
+    if ($verbose == true) then
+      puts message
+    end
+end
+
+def checkout_pull_request(pr, target_directory)
+    FileUtils.mkdir_p(".vuepress")
+
+    pull_request_url = "https://api.github.com/repos/openhab/openhab-docs/pulls/#{pr}"
+    
+    response = JSON.parse(open(pull_request_url).read)
+    repository_url =  response['head']['repo']['clone_url']
+    label = response['head']['label']
+    sha = response['head']['sha']
+    branch = response['head']['ref']
+    title = response['title']
+
+    puts "➡️ Cloning repository 📦 #{label} ..."
+    puts "  ↪️ PR ##{pr}: #{title}"
+
+    system("OH_DOCS_VERSION=#{branch}")
+  
+    FileUtils.cd(target_directory, verbose: false) do
+      system("git clone --depth 1 #{repository_url} --branch #{branch} #{$verbose ? '' : '--quiet'}")
+      system("git reset ##{sha} #{$verbose ? '' : '--quiet'}")
+    end
+  end
+
+verbose "🧹 Cleaning existing documentation downloads ..."
```

Comment:
I love the broom here <3

Reply:
I'm getting old - and using emoji's helps a bit against this feeling. But I know that probably this is not state of the art anymore with the next generation. But it adds a nice touch to the outputs ... (opinionated - as usual 😉).